JavaScript/ds/Hash.js

```diff
@@ -0,0 +1,172 @@
+    // Item is not present in hash table
+    if (item === undefined)
+      return console.log("Can't delete, no such item found");
+
+    // Delete item present at that location from hash table
+    delete this.data[item.location];
+    return console.log("Deleted", item.data[0]);
+  }
+
+  // Display all items
+  keys() {
+    const keysArray = [];
+    for (let i = 0; i < this.data.length; i++) {
+      if (this.data[i]) keysArray.push(this.data[i][0][0]);
+    }
+
+    return keysArray;
+  }
+}
+
+// Instantiating the HashTable
+const myHashTable = new HashTable(100);
+
+// Insert item in Hash Table
+myHashTable.set("grapes", 1000); // Output - [['grapes', 1000]]
+myHashTable.set("apple", 100); // Output - [['apple', 100], ['grapes', 1000]]
+myHashTable.set("oranges", 50); // Output - [['apple', 100], ['grapes', 1000], ['oranges', 50]]
+
+myHashTable.deleteItem("apple"); // Output - [['grapes', 1000], ['oranges', 50]]
+myHashTable.deleteItem("fruits"); // Output - Can't delete, no such item found
+
+let choice = 0;
+let value, index;
+
+// Asking user input for performing operations on Hash Table
+do {
+  console.log("\nWelcome to the Hash Table");
+  console.log("Follow the instructions to get in the show");
+  console.log("0. To exit without doing anything".yellow);
+  console.log("1. Display all elements of the Hash Table".display);
+  console.log("2. Insert element in the Hash Table".insert);
+  console.log("3. Find element in Hash Table".display);
+  console.log("4. Delete element from Hash Table".delete);
+
+  choice = +prompt("Enter your choice - ");
+
+  switch (choice) {
+    case 0:
+      console.log("Hmm, matter of choice!\n".rainbow);
+      return;
+
+    case 1:
+      console.log("Hash Table - ", myHashTable.keys());
+      break;
+
+    case 2:
+      value = prompt("Enter Value ( first value )  to Insert element - ");
+      index = +prompt("Enter amount ( second value) to Insert element - ");
+
+      if (index <= 0)
+        return console.log("You know this is wrong, Bbye!\n".wrong);
+
+      myHashTable.set(value, index);
+      console.log("After Insertion Hash Table - ", myHashTable.keys());
+      break;
+
+    case 3:
+      if (myHashTable.data.length === 0) {
+        return console.log("Hash Table in empty".wrong);
+      }
+
+      value = prompt("Enter Value to find the element - ");
+      let result = myHashTable.get(value);
+
+      if (!result) console.log("Element not found");
+      else console.log("Element ", result);
+      break;
+
+    case 4:
+      if (myHashTable.data.length === 0) {
+        return console.log(
+          "404, you can't remove anything from empty hash table".wrong
+        );
+      }
+
+      value = prompt("Enter Value to delete - ");
+      myHashTable.deleteItem(value);
+      console.log("After Deletion Hash Table - ", myHashTable.keys());
+      break;
+
+    default:
+      console.log("You know this is wrong, Bbye!\n".wrong);
+      return;
+  }
+} while (choice !== 0);
```

Comment:
Add sample I/O at the end of the code

Reply:
Added sample I/O
Let me know about further changes @HarshCasper 

P.S. If the I/O scenario that I have created is good, I will like to do it for other cases as well  ( Ex., Linked List in JS )